src/symbol/static_graph.cc

```diff
@@ -161,6 +161,97 @@ bool StaticGraph::InferNodeShapes(const std::vector<uint32_t> &topo_order,
   return true;
 }
 
+bool StaticGraph::InferNodeTypes(const std::vector<uint32_t> &topo_order,
```

Comment:
I thought about this since I only needed to change shape to type and the rest of the code is untouched. But I didn't want to change the existing code too much.